c7n/filters/vpc.py

```diff
@@ -173,9 +173,10 @@ def process(self, resources, event=None):
         results = []
         for r in resources:
             resource_sgs = self.filter_ignored(
-                [related_sg[sid] for sid in self.sg.get_related_ids([r])])
-            resource_subnets = self.filter_ignored([
-                related_subnet[sid] for sid in self.subnet.get_related_ids([r])])
+                [related_sg[sid] for sid in self.sg.get_related_ids([r]) if sid in related_sg])
```

Comment:
here @kapilt this change looks legit since related_sg and realted_subnets( in the following line) are loading up all sgs/subnets but the ids returned by get_related_ids might not be present if they get deleted.